src/main/kotlin/com/cosmotech/api/containerregistry/ContainerRegistryService.kt

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.containerregistry
+
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.exceptions.CsmResourceNotFoundException
+import java.util.Base64
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.client.methods.RequestBuilder
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.springframework.http.HttpHeaders
+import org.springframework.stereotype.Service
+
+@Service("csmContainerRegistry")
+class ContainerRegistryService(private val csmPlatformProperties: CsmPlatformProperties) {
+  fun getEndpoint() = csmPlatformProperties.containerRegistry.registryUrl
+
+  fun getCredentials(user: String, password: String) =
+      Base64.getEncoder().encodeToString("$user:$password".toByteArray())
+
+  fun getHeaderAuthorization() =
+      "Basic " +
+          getCredentials(
+              csmPlatformProperties.containerRegistry.registryUserName!!,
+              csmPlatformProperties.containerRegistry.registryPassword!!)
+
+  private fun buildHttpRequest(repository: String): HttpUriRequest {
+    var reqBuilder = RequestBuilder.get()
+    reqBuilder = reqBuilder.setUri(getImageRegistryUri(repository))
+    reqBuilder = reqBuilder.addHeader(HttpHeaders.AUTHORIZATION, getHeaderAuthorization())
+
+    return reqBuilder.build()
+  }
+
+  fun checkSolutionImage(repository: String, tag: String) {
+    val tags = getRepositoryTagList(repository)
+    if (tags.contains("errors") || !doesTheTagExist(tags, tag)) {
```

Comment:
What happens if a user tags its solution `mysolution_simulator:1.2.3-errors`? Shouldn't we use the HTTP return code to check for errors instead of greping the response?